scaleway/resource_domain_record_test.go

```diff
@@ -0,0 +1,606 @@
+package scaleway
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+var (
+	reservedDomains = []*regexp.Regexp{
+		regexp.MustCompile(`.*iliad.*`),
+		regexp.MustCompile(`.*\.free\..*`),
+		regexp.MustCompile(`.*\.online\..*`),
+		regexp.MustCompile(`.*scaleway.*`),
+		regexp.MustCompile(`.*dedibox.*`),
+	}
+	testDomain = ""
+)
+
+func init() {
+	// retrieve the first domain is found
+	scwClient, err := sharedClientForZone(scw.Zone(""))
+	if err != nil {
+		l.Errorf("error when retrieve scwClient: %s", err)
+		return
+	}
+
+	domainRegistrarAPI := domain.NewRegistrarAPI(scwClient)
+	listDomains, err := domainRegistrarAPI.ListDomains(&domain.RegistrarAPIListDomainsRequest{
+		Status:     domain.DomainStatusActive,
+		OrderBy:    domain.ListDomainsRequestOrderByDomainAsc,
+		IsExternal: scw.BoolPtr(true),
+	})
+	if err != nil {
+		l.Errorf("error listing domains: %s", err)
+		return
+	}
+
+	for _, domain := range listDomains.Domains {
+		// skipping reserved domains
+		isReserved := false
+		for _, reservedDomain := range reservedDomains {
+			if reservedDomain.MatchString(domain.Domain) {
+				isReserved = true
+				break
+			}
+		}
+
+		if !isReserved {
+			testDomain = domain.Domain
+			break
+		}
+	}
+
+	if testDomain == "" {
+		l.Errorf("your account doesn't have any domain to test with")
+		return
+	}
+
+	l.Infof("start domain record test with domain: %s", testDomain)
+}
+
+func TestAccScalewayDomainRecord_Basic(t *testing.T) {
+	tt := NewTestTools(t)
+	// Please note: don't use `defer tt.Cleanup()` to avoid store personnal domain in cassette testdata
```

Comment:
We could implement a filter, the same way that we filter out secret key from cassettes. This could be a solution, we would only need to have a domain name to run those tests. This domain could be passed by an environment variable to be easily configured in a CI

Reply:
I've added a new env var "TF_TEST_DOMAIN" to force using a domain to test with.